src/calendarmanager.cpp

```diff
@@ -520,7 +517,8 @@ void CalendarManager::doAgendaAndQueryRefresh()
         }
     }
 
-    if (!missingRanges.isEmpty() || !missingInstanceList.isEmpty()) {
+    if ((!missingRanges.isEmpty() || !missingInstanceList.isEmpty())
+        && !mLoadPending) {
```

Comment:
> But one corner case I'm left pondering is the resetPending which does some side-effects above 

Ah indeed, very nice catch. Yes, the side effects need to be adressed otherwise, running twice the doAgendaRefresh() with reset pending during a long load pending in progress, would not have called for the same events...

As you suggest, I propose to move the side effects later, when the data are actually loaded. I think it's safe. For the queries, there is already `|| resetPending` in the if, ensuring that all current queries are requested during the `loadData()` call. And for the range, I'm keeping the `if resetPending` that adds the loaded ranges. About the data themselves (events), they are safe in the manager as long as the `dataLoadedSlot` is not called with the reset argument to true. So the manager can rely on loadedRanges and loadedQueries as long as the callback is not triggered.